### ISSUE TITLE
Trying to get pod logs before creation start results in error screen

### ISSUE THREAD

**Issue by LarryGF** [argoproj/argo-cd#13575]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

From the UI, when trying to get the logs of a Deployment/Pod, and the pod has not started its creation process, the UI crashes with the error shown below. In order for the error to go away it is necessary to reload the page.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Version**

```shell
argocd: v2.7.1+5e54351.dirty
  BuildDate: 2023-05-02T16:35:40Z
  GitCommit: 5e543518dbdb5384fa61c938ce3e045b4c5be325
  GitTreeState: dirty
  GoVersion: go1.19.6
  Compiler: gc
  Platform: linux/amd64
```

**Logs**

```
Stacktrace:

TypeError: Reduce of empty array with no initial value
    at Array.reduce (<anonymous>)
    at Object.children (http://localhost:8080/main.a726808c41da12ea7cc9.js:2:1885533)
    at Ji.render (http://localhost:8080/main.a726808c41da12ea7cc9.js:2:1550799)
    at Hs (http://localhost:8080/main.a726808c41da12ea7cc9.js:2:820265)
    at js (http://localhost:8080/main.a726808c41da12ea7cc9.js:2:820060)
    at ya (http://localhost:8080/main.a726808c41da12ea7cc9.js:2:855541)
    at pl (http://localhost:8080/main.a726808c41da12ea7cc9.js:2:847016)
    at ul (http://localhost:8080/main.a726808c41da12ea7cc9.js:2:846941)
    at il (http://localhost:8080/main.a726808c41da12ea7cc9.js:2:843971)
    at http://localhost:8080/main.a726808c41da12ea7cc9.js:2:795750
```

**Comment by patst** [comment]
looks similar: https://github.com/argoproj/argo-cd/issues/13447

**Comment by diclophis** [comment]
I am running into this issue, it seems that it only happens on certain container names?

**Comment by alexmt** [comment]
Issue is fixed in v2.7.3
